typed/comparison_test.go

```diff
@@ -0,0 +1,185 @@
+package typed_test
+
+import (
+	"testing"
+
+	"sigs.k8s.io/structured-merge-diff/v3/fieldpath"
+	"sigs.k8s.io/structured-merge-diff/v3/typed"
+)
+
+func TestComparisonRemove(t *testing.T) {
```

Comment:
I know this sounds weird, but there are too many tests here. "Compare" is merely applying `RecursiveDifference` to three different sets. You should have a test to make sure that you apply it to all three sets and a test to check the empty set and that's it. 

On the other hand, you don't have any tests for `RecursiveDifference`, which is not good :-) And you should add that to the benchmark suite too.

Reply:
Moved most of the tests to the set_test and added a benchmark.
It's a little worse than the normal Difference, because of the additional call(s) to s2.Members.Has (i assume).